mediorum/server/serve_image.go

```diff
@@ -22,14 +24,31 @@ func (ss *MediorumServer) serveImage(c echo.Context) error {
 	jobID := c.Param("jobID")
 	variant := c.Param("variant")
 	isOriginalJpg := variant == "original.jpg"
+	cacheKey := jobID + variant
 
-	// helper function... only sets cache-control header on success
-	serveSuccessWithReader := func(blob *blob.Reader) error {
+	serveSuccessWithBytes := func(blobData []byte, modTime time.Time) error {
 		c.Response().Header().Set(echo.HeaderCacheControl, "public, max-age=2592000, immutable")
-		http.ServeContent(c.Response(), c.Request(), jobID+variant, blob.ModTime(), blob)
+		http.ServeContent(c.Response(), c.Request(), cacheKey, modTime, bytes.NewReader(blobData))
 		return nil
 	}
 
+	// helper function... only sets cache-control header on success
+	serveSuccessWithReader := func(blob *blob.Reader) error {
+		blobData, err := io.ReadAll(blob)
+		if err != nil {
+			return err
+		}
+		blob.Close()
+		ss.imageCache.Set(cacheKey, blobData, imcache.WithNoExpiration())
```

Comment:
so no expiration just means the cache will drop things that havent been hit in a while?

Reply:
it's LRU style so the recent thing bumps the old thing

Reply:
we talked about this and wished there was a "max mem" function in the lib, but this should cover us quite well